test/commonControlTests/NumberControl.spec.ts

```diff
@@ -103,6 +158,46 @@ suite('NumberControl e2e tests', () => {
                         .value,
                 ).eq(16);
             });
+
+            test('first number via screen not valid, second number via screen valid', async () => {
+                const requestHandler = new ControlHandler(new NumberControlManager());
+                await testE2E(requestHandler, [
+                    'U: ',
+                    TestInput.of(GeneralControlIntent.of({})),
+                    'A: What number?',
+                    'U: ',
+                    TestInput.simpleUserEvent(['NumberSelectorWithoutValidationExpectation', 15]),
+                    "A: Sorry but that's not a valid choice because the value must be even. What number?",
+                    'U: ',
+                    TestInput.simpleUserEvent(['NumberSelectorWithoutValidationExpectation', 16]),
+                    'A: Ok. Value set to 16.',
+                ]);
+                expect(
+                    (requestHandler.getSerializableControlStates()[TEST_CONTROL_ID] as NumberControlState)
+                        .value,
+                ).eq(16);
+            });
+
+            test('first number via screen not valid, second number via voice valid', async () => {
+                const requestHandler = new ControlHandler(new NumberControlManager());
+                await testE2E(requestHandler, [
+                    'U: ',
+                    TestInput.of(GeneralControlIntent.of({})),
+                    'A: What number?',
+                    'U: ',
+                    TestInput.simpleUserEvent(['NumberSelectorWithoutValidationExpectation', 15]),
+                    "A: Sorry but that's not a valid choice because the value must be even. What number?",
+                    'U: 16',
+                    TestInput.of(
+                        SingleValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, { 'AMAZON.NUMBER': '16' }),
+                    ),
+                    'A: Ok. Value set to 16.',
```

Comment:
shouldn't it be `Was that 16?` to confirm since it was voice?

Reply:
In this case confirmation is only required if the value < 10 or > 20, but in general, yes we do confirm when values are given by voice (other tests verify this).